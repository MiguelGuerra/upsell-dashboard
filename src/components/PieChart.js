import React from 'react'
import { Doughnut } from 'react-chartjs-2'

function PieChart({
    labelsArray,
    caption,
    dataArray}) {
    const data = {
        labels: labelsArray,
        datasets: [
            {
                label: caption,
                data: dataArray,
                backgroundColor: [
                    '#ffae4a',
                    '#41a6c3',
                    '#fb928E'
                ]
            }
        ]
    }
    return (
        <div className="pie-chart">
            <Doughnut
                data={data}
                options={{ maintainAspectRatio: false }} />
        </div>
    )
}

export default PieChart
