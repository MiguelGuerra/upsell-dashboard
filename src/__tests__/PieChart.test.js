import React from 'react'
import renderer from 'react-test-renderer'
import PieChart from '../components/PieChart'

it("PieChart renders when is receives props", () => {
    const tree = renderer.create(
        <PieChart
            labelsArray={['Emails', 'Reservations', 'Transactions']}
            caption="Results on month of September"
            dataArray={['45', '50']}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})