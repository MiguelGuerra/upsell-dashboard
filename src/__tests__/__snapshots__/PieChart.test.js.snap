// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart renders when is receives props 1`] = `
<div
  className="pie-chart"
  data-testid="pie-chart"
>
  <canvas
    data-testid="canvas"
    height={150}
    onClick={[Function]}
    role="img"
    width={300}
  />
</div>
`;
